### PR TITLE
DR-1829 Add search unit tests

### PR DESCRIPTION
**Pull request by fboulnois**
Add unit tests for indexing endpoint.

### CODE REVIEW

File: src/test/java/bio/terra/service/search/SearchTest.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        return new SearchIndexRequest().sql(sqlQuery);
+    }
+
+    private List<Map<String, Object>> getSnapshotTableData() {
+        List<Map<String, Object>> values = new ArrayList<>();
+        for (int i = 0; i < 3; i++) {
+            Map<String, Object> row = new HashMap<>();
+            row.put("uuid", UUID.randomUUID().toString());
+            row.put("now", Instant.now().toString());
+            values.add(row);
```
Comment (pshapiro4broad) [inline src/test/java/bio/terra/service/search/SearchTest.java]: Can use `Map.of()`
```suggestion
            values.add(Map.of("uuid", UUID.randomUUID().toString(), "now", Instant.now().toString());
```

File: src/test/java/bio/terra/service/search/SearchTest.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    }
+
+    @Test
+    public void indexSnapshotTest() throws Exception {
+        mockGetSnapshotTableData();
+        //service.indexSnapshot(snapshot, searchIndexRequest);
+    }
+
+    private void mockGetSnapshotTableData() throws InterruptedException {
+        when(bigQueryPdao.getSnapshotTableData(snapshot, searchIndexRequest.getSql()))
```
Comment (pshapiro4broad) [inline src/test/java/bio/terra/service/search/SearchTest.java]: Looking at the code, we will probably want a unit test for `BigQueryPdao.getSnapshotTableData()` too. (In `BigQueryPdaoTest`, unless you want to make another test class that only has `BigQueryPdao` unit tests.)

File: src/test/java/bio/terra/service/search/SearchTest.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    private List<Map<String, Object>> values;
+
+    @Before
+    public void setup() throws Exception {
+        MockitoAnnotations.initMocks(this);
+        service = new SearchService(bigQueryPdao, client);
+
+        searchIndexRequest = getSearchIndexRequest();
+        snapshot = getSnapshot();
+        values = getSnapshotTableData();
```
Comment (pshapiro4broad) [inline src/test/java/bio/terra/service/search/SearchTest.java]: We do this in a lot of our tests, but one thing to think about is that `@Before` is called before every test, and it's better to only put things in it that need to be reinitialized before every test. These objects appear to be static so they could be done in a `@BeforeClass`.

File: src/test/java/bio/terra/service/search/SearchTest.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.mockito.Mockito.when;
+
+@Category(Unit.class)
+public class SearchTest {
```
Comment (pshapiro4broad) [inline src/test/java/bio/terra/service/search/SearchTest.java]: I'm not sure what the current conventions are, but I would expect a test class's name to reflect the class it's testing. So I would expect this to be called `SearchServiceTest`.

File: src/test/java/bio/terra/service/search/SearchTest.java (hunk 3 above)
Comment (fboulnois) [reply src/test/java/bio/terra/service/search/SearchTest.java]: I tried using a `@BeforeClass` , but I get the following error:

```
> No tests found for given includes: [bio.terra.service.search.SearchServiceTest.indexSnapshotTest](filter.includeTestsMatching)
```

Comment (pshapiro4broad) [reply src/test/java/bio/terra/service/search/SearchTest.java]: The `BeforeClass` method needs to be public and static, maybe that was the issue? https://junit.org/junit4/javadoc/4.13/org/junit/BeforeClass.html

File: src/test/java/bio/terra/service/search/SearchTest.java (hunk 1 above)
Comment (fboulnois) [reply src/test/java/bio/terra/service/search/SearchTest.java]: 👍

File: src/test/java/bio/terra/service/search/SearchTest.java (hunk 2 above)
Comment (fboulnois) [reply src/test/java/bio/terra/service/search/SearchTest.java]: 👍

File: src/test/java/bio/terra/service/search/SearchTest.java (hunk 4 above)
Comment (fboulnois) [reply src/test/java/bio/terra/service/search/SearchTest.java]: 👍

File: src/test/java/bio/terra/service/search/SearchTest.java (hunk 3 above)
Comment (fboulnois) [reply src/test/java/bio/terra/service/search/SearchTest.java]: `BeforeClass` seems to be interacting poorly with Mockito. If I set the method to static, then all the member variables assigned in the constructor need to also be static, but this causes an error with the `@Mock` annotation.

Comment (pshapiro4broad) [reply src/test/java/bio/terra/service/search/SearchTest.java]: Ah well, never mind then. Thanks for trying it out.

File: src/test/java/bio/terra/service/search/SearchServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+
+    @Test
+    public void indexSnapshotTest() throws Exception {
+        mockGetSnapshotTableData();
+        mockIndexRequest();
+        mockIndexResponse();
+        service.indexSnapshot(snapshot, searchIndexRequest);
+    }
+
+    private void mockGetSnapshotTableData() throws InterruptedException {
```
Comment (fboulnois) [inline src/test/java/bio/terra/service/search/SearchServiceTest.java]: I'm fine with moving these back into the test if it's unclear.

File: src/test/java/bio/terra/service/search/SearchServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+        values = getSnapshotTableData();
+    }
+
+    @Test
+    public void indexSnapshotTest() throws Exception {
+        mockGetSnapshotTableData();
+        mockIndexRequest();
+        mockIndexResponse();
+        service.indexSnapshot(snapshot, searchIndexRequest);
+    }
```
Comment (pshapiro4broad) [inline src/test/java/bio/terra/service/search/SearchServiceTest.java]: I think a test needs at least one `assert()`. Is there something that can be asserted here to ensure that code did something? If all we can do is assert that some API was called with some arguments, you can use `Mockito.verify()` to do that.

Comment (fboulnois) [reply src/test/java/bio/terra/service/search/SearchServiceTest.java]: Good point. I think I can assert the response.

Comment (fboulnois) [reply src/test/java/bio/terra/service/search/SearchServiceTest.java]: I added this.

**Review by pshapiro4broad — APPROVED** [review]
LGTM!

**Review by tcjiang9 — APPROVED** [review]

**Pull request merged**